app/api/resources/user.py

```diff
@@ -498,3 +498,111 @@ def get(cls):
+    @users_ns.response(HTTPStatus.OK, f"{messages.PASSWORD_RESET_MAIL_MESSAGE}")
+    @users_ns.response(HTTPStatus.BAD_REQUEST, f"{messages.EMAIL_FIELD_IS_MISSING}")
+    @users_ns.response(HTTPStatus.FORBIDDEN, f"{messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN}")
+    @users_ns.response(HTTPStatus.NOT_FOUND, f"{messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM}")
+    @users_ns.expect(forgot_password_change_request_data_model, validate=True)
+ 
+    def post(self):
+        """ Updates User's password by sending reset link to email
+ 
+        Args: 
+          email : Verified email of the user
+ 
+        Return:
+          token: a token link/reset link will be shared on verified mail of user
+        """
+ 
+        data = request.json
+ 
+        if not data:
+            return messages.EMAIL_FIELD_IS_MISSING, HTTPStatus.BAD_REQUEST
+ 
+        is_valid = validate_forgot_password_email(data)
+ 
+        if is_valid!={}:
+            return is_valid, HTTPStatus.BAD_REQUEST
+        
+        user = DAO.get_user_by_email(data["email"])
+ 
+        if user is None:
+            return messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM, HTTPStatus.NOT_FOUND
+ 
+        if not user.is_email_verified:
+            return messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN, HTTPStatus.FORBIDDEN    
+ 
+        send_email_reset_password_message(user.name, data["email"])
+ 
+        return messages.PASSWORD_RESET_MAIL_MESSAGE, HTTPStatus.OK
+ 
+@users_ns.route("user/reset_password")
+@users_ns.response(HTTPStatus.OK, f"{messages.PASSWORD_SUCCESSFULLY_UPDATED}")
+@users_ns.response(
+    HTTPStatus.UNAUTHORIZED,
+    "%s\n%s\n%s"
+    % (
+        messages.TOKEN_HAS_EXPIRED,
+        messages.TOKEN_IS_INVALID,
+        messages.AUTHORISATION_TOKEN_IS_MISSING,
+    ),
+)
+@users_ns.response(HTTPStatus.BAD_REQUEST, f"{messages.PASSWORD_FIELD_IS_MISSING}")    
+ 
+class ResetPassword(Resource):
+    @classmethod
+    @users_ns.doc(f"reset")
+    @users_ns.doc(
+        responses={
+            HTTPStatus.UNAUTHORIZED: f"{messages.TOKEN_HAS_EXPIRED['message']}<br>"
+            f"{messages.TOKEN_IS_INVALID['message']}<br>"
+            f"{messages.AUTHORISATION_TOKEN_IS_MISSING['message']}"
+        }
+    )
+    @users_ns.expect(reset_password_forgot_request_data_model, validate=True)
+ 
+    def post(self):
+        """ Updates User's password
+ 
+        This will allow the user to change/create a new password 
+        In case they forgot it.
+ 
+        Args:
+        token: token of reset link sent on verified mail
+        new_password: take the new validated password to login
+ 
+        Return:
+        Password Updated successfully
+        """
+        data = request.json
+       
+        if not data["token"]:
+            return messages.AUTHORISATION_TOKEN_IS_MISSING, HTTPStatus.UNAUTHORIZED
+ 
+        if not data["new_password"] :
+            return messages.PASSWORD_FIELD_IS_MISSING, HTTPStatus.BAD_REQUEST    
```

Comment:
Maybe we can add a new message for password_field_is_empty or something like that :sweat_smile: and this can be moved into `validate_reset_password`

Reply:
as @isabelcosta said we were using http status code so, we should use this with the required messages. also, the password was still required so, i don't think to create a new message for password missing is required as one message is already there for it and it may make a redundancy in the code . what do you think?